tokio-trace/src/lib.rs

```diff
@@ -49,153 +49,7 @@
-//! ```
-//! # #[macro_use]
-//! # extern crate tokio_trace;
-//! # use tokio_trace::Level;
-//! # fn main() {
-//! let user = "ferris";
-//!
-//! span!(Level::TRACE, "login", user);
-//! // is equivalent to:
-//! span!(Level::TRACE, "login", user = user);
-//! # }
-//!```
-//!
-//! The [`field::display`] and [`field::debug`] functions are used to record
-//! fields on spans or events using their `fmt::Display` and `fmt::Debug`
-//! implementations (rather than as typed data). This may be used in lieu of
-//! custom `Value` implementations for complex or user-defined types.
-//!
-//! In addition, the span and event macros permit the use of the `%` and `?`
-//! sigils as shorthand for `field::display` and `field::debug`, respectively.
-//! For example:
-//!
-//! ```
-//! # #[macro_use]
-//! # extern crate tokio_trace;
-//! # use tokio_trace::{Level, field};
-//! # fn main() {
-//! #[derive(Debug)]
-//! struct MyStruct {
-//!     my_field: &'static str,
-//! }
-//!
-//! let my_struct = MyStruct {
-//!     my_field: "Hello world!"
-//! };
-//!
-//! span!(Level::TRACE,"my_span", ?my_struct, %my_struct.my_field);
-//! // is equivalent to:
-//! span!(Level::TRACE, "my_span", my_struct = field::debug(&my_struct), my_struct.my_field = field::display(&my_struct.my_field));
-//! # }
-//!```
-//!
-//! ### When to use spans
-//!
-//! As a rule of thumb, spans should be used to represent discrete units of work
-//! (e.g., a given request's lifetime in a server) or periods of time spent in a
-//! given context (e.g., time spent interacting with an instance of an external
-//! system, such as a database).
-//!
-//! Which scopes in a program correspond to new spans depend somewhat on user
-//! intent. For example, consider the case of a loop in a program. Should we
-//! construct one span and perform the entire loop inside of that span, like:
-//! ```rust
-//! # #[macro_use] extern crate tokio_trace;
-//! # use tokio_trace::Level;
-//! # fn main() {
-//! # let n = 1;
-//! let span = span!(Level::TRACE, "my_loop");
-//! let _enter = span.enter();
-//! for i in 0..n {
-//!     # let _ = i;
-//!     // ...
-//! }
-//! # }
-//! ```
-//! Or, should we create a new span for each iteration of the loop, as in:
-//! ```rust
-//! # #[macro_use] extern crate tokio_trace;
-//! # use tokio_trace::Level;
-//! # fn main() {
-//! # let n = 1u64;
-//! for i in 0..n {
-//!     let span = span!(Level::TRACE, "my_loop", iteration = i);
-//!     let _enter = span.enter();
-//!     // ...
-//! }
-//! # }
-//! ```
-//!
-//! Depending on the circumstances, we might want to do either, or both. For
-//! example, if we want to know how long was spent in the loop overall, we would
-//! create a single span around the entire loop; whereas if we wanted to know how
-//! much time was spent in each individual iteration, we would enter a new span
-//! on every iteration.
+//! The [`span` module]'s documentation provides further details on how to use spans.
 //!
 //! ## Events
```

Comment:
This section reinforces the belief that a span establishes a "context", which again does not match my understand of what Spans actually mean?

Reply:
Further down, this text also says:
> Events are represented as a special case of spans — they are created, they may have fields added, and then they close immediately, without being entered.

Is that true? I thought events _weren't_ just "immediate spans"?

Reply:
> Is that true? I thought events weren't just "immediate spans"?

That sentence is no longer correct, I think it was written when this _was_ the case and then it was never removed. I must've missed that while working on this branch, since I was mostly just looking at the span docs. Thanks for catching that!